server/controllers/finance/account.js

```diff
@@ -104,6 +105,35 @@ function detail(req, res, next) {
   .done();
 }
 
+function getBalance (req, res, next){
+  'use strict';
+
+  var accountId = req.params.id, optional = '';
+  var params = [accountId];
+
+  if(req.query.journal === '1') {
+    optional = ' UNION ALL SELECT pj.account_id, IFNULL(SUM(pj.debit), 0) AS debit, IFNULL(SUM(pj.credit), 0) AS credit, IFNULL((pj.debit - pj.credit), 0) AS balance FROM posting_journal AS pj WHERE pj.account_id = ? GROUP BY pj.account_id';
+    params.push(accountId);
+  }
+
+  var accountSoldQuery = 'SELECT t.account_id, IFNULL(SUM(t.debit), 0) AS debit, IFNULL(SUM(t.credit), 0) AS credit, IFNULL(t.balance, 0) AS balance FROM' + 
+    ' (SELECT gl.account_id, IFNULL(SUM(gl.debit), 0) AS debit, IFNULL(SUM(gl.credit), 0) AS credit, IFNULL((gl.debit - gl.credit), 0) AS balance FROM' + 
+    ' general_ledger AS gl WHERE gl.account_id = ? GROUP BY gl.account_id' + optional + ' ) AS t GROUP BY t.account_id';
+
+  lookupAccount(accountId, req.codes)
+    .then(function (account) {
+      return db.exec(accountSoldQuery, params);
+    })
+    .then(function (rows){
+      if(rows.length === 0){
+        res.status(200).json({account_id : accountId, debit : 0, credit : 0, balance : 0});
+      }
+      res.status(200).json(rows[0]);
```

Comment:
Is there a test case for this?  I'm pretty sure express will throw an error here if `rows.length === 0`.  The error will be [cannot set headers after they are sent](http://stackoverflow.com/questions/26307920/res-json-cant-set-headers-after-they-are-sent), since you will attempt to call `res.json()` twice.

To fix it, just add a `return` before the first `res.status(200)` call.  In the previous code, it was a `throw` which implicitly returns.

To avoid this in the future, a clearer way to write would be to ensure `res.status(200).json()` is only called once, no matter the scenario.  For example:
```js
.then(function (rows) {

  // if there is no result, send back a zero balance.  Otherwise, send back the result
  var response = (rows.length === 0) ?
       {account_id : accountId, debit : 0, credit : 0, balance : 0 } :
       rows[0];

  res.status(200).json(response);
});
```